Rakefile

```diff
@@ -39,8 +39,6 @@ rescue LoadError
   end
 end
 
-task :test => :check_dependencies
```

Comment:
Was this ever working? Any idea what it should do?

Reply:
Yes, it was and still is working:
```
$ rake test
/usr/local/Cellar/ruby/2.4.1_1/bin/ruby -w -I"lib:lib:test" -I"/usr/local/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib" "/usr/local/lib/ruby/gems/2.4.0/gems/rake-12.0.0/lib/rake/rake_test_loader.rb" "test/test_canonicalise.rb" "test/test_compile.rb" "test/test_decompile.rb" "test/test_play.rb" "test/test_playadams.rb" "test/test_save.rb" 
All dependencies seem to be installed.
skipping 'tutorial/t7.sck' because no filename
............

Finished in 2.271812 seconds.
------------------------------------------------------------------------------------------------------------------------------------
12 tests, 56 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
------------------------------------------------------------------------------------------------------------------------------------
5.28 tests/s, 24.65 assertions/s
```

So I don't want to lose this line. What make you think it wasn't working?

Reply:
Interesting, when I run it on this branch with the Gemfile I get:
```
% bundle exec rake
rake aborted!
Don't know how to build task 'check_dependencies' (see --tasks)
/Users/andrew/.rbenv/versions/2.3.1/bin/bundle:23:in `load'
/Users/andrew/.rbenv/versions/2.3.1/bin/bundle:23:in `<main>'
Tasks: TOP => default => test
(See full trace by running task with --trace)
% rake
rake aborted!
Don't know how to build task 'check_dependencies' (see --tasks)

Tasks: TOP => default => test
(See full trace by running task with --trace)
```
But on master it works as you've demonstrated.

Looking around I found it looks like [it's part of Jeweler](http://www.rubydoc.info/github/technicalpickles/jeweler/Jeweler/Commands/CheckDependencies) so it must be something odd with the require order?

In any case it doesn't look like something they're using anymore. I generated a new gem following their instructions and this is the Rakefile:
```rb
# encoding: utf-8

require 'rubygems'
require 'bundler'
begin
  Bundler.setup(:default, :development)
rescue Bundler::BundlerError => e
  $stderr.puts e.message
  $stderr.puts "Run `bundle install` to install missing gems"
  exit e.status_code
end
require 'rake'

require 'jeweler'
Jeweler::Tasks.new do |gem|
  # gem is a Gem::Specification... see http://guides.rubygems.org/specification-reference/ for more options
  gem.name = "hello-gem"
  gem.homepage = "http://github.com/drewish/hello-gem"
  gem.license = "MIT"
  gem.summary = %Q{TODO: one-line summary of your gem}
  gem.description = %Q{TODO: longer description of your gem}
  gem.email = "drewish@katherinehouse.com"
  gem.authors = ["andrew morton"]
  # dependencies defined in Gemfile
end
Jeweler::RubygemsDotOrgTasks.new

require 'rake/testtask'
Rake::TestTask.new(:test) do |test|
  test.libs << 'lib' << 'test'
  test.pattern = 'test/**/test_*.rb'
  test.verbose = true
end

desc "Code coverage detail"
task :simplecov do
  ENV['COVERAGE'] = "true"
  Rake::Task['test'].execute
end

task :default => :test

require 'rdoc/task'
Rake::RDocTask.new do |rdoc|
  version = File.exist?('VERSION') ? File.read('VERSION') : ""

  rdoc.rdoc_dir = 'rdoc'
  rdoc.title = "hello-gem #{version}"
  rdoc.rdoc_files.include('README*')
  rdoc.rdoc_files.include('lib/**/*.rb')
end
```

Reply:
Predictably -- and this kind of thing underlies my ongoing frustration with the Ruby community as a whole -- the new Rakefile is _totally_ different from the old one.

Let's ignore that for now, revisiting it if we find that we need to.

Anyway, I now see that if you remove the `task :test => :check_dependencies` line, as you wish, `rake test` still works fine -- apparently, exactly as it did before. So, sure, we can remove it.